### ISSUE TITLE
Use return codes to communicate checking status

### ISSUE THREAD

**Issue by madig** [googlefonts/fontbakery#1759]
## Observed behaviour

Apparently fontbakery always returns zero even with errors and failures.

## Expected behaviour

Errors and failures during a test should make the return code > 0.

**Comment by graphicore** [comment]
We should probably add a generic interface to `FontbakeryReporter`. It already registers the statuses, so any reporter used with `runner.run` could be used to get the end result.

`FontbakeryReporter._ended` is set to the value of the `END` event, like:

```py
myFontbakeryReporter._ended = event = (END, checkrun_summary, (None, None, None))
```

and `checkrun_summary` is a `collections.Counter` where the keys are amongst the others statuses are `ERROR` and `FAIL`.


thus something like:

```py
# fontbakery/reporters/__init__.py

# …
check_statuses = [ERROR, FAIL, SKIP, PASS, WARN, INFO]
check_statuses.sort(key=lambda s:s.weight, reverse=True)

# …

Class class FontbakeryReporter(object):
# ...
  def end_status(self):
    if not self._ended:
      return None
    _, summary, _ = self._ended
    for status in check_statuses:
      if status.name in summary and summary[status.name]:
        return status
    # Raise or warn?
    return None
```

then in main we can do something like

```py
# fontbakery/commands/check_spec.py

def main(specification=None, values=None):
  # ...
  endStatus = tr.end_status()
  return 1 if endStatus in (ERROR, FAIL) else 0
```

**Comment by graphicore** [comment]
Fixed in #1804
